apps/site/providers/modalProvider.tsx

```diff
@@ -0,0 +1,56 @@
+'use client';
+
+import type { FC, PropsWithChildren, ComponentType } from 'react';
+import { createContext, useState } from 'react';
+
+import ReleaseModal from '../components/Downloads/ReleaseModal';
+import EOLModal from '../components/EOL/Modal';
+
+export type ModalProps = {
+  open: boolean;
+  closeModal: () => void;
+  data: unknown;
+};
+
+type ModalContextType = {
+  data: unknown;
+  openModal: (data: unknown) => void;
+  closeModal: () => void;
+};
+
+export const ModalContext = createContext<ModalContextType>({
+  data: null,
+  openModal: () => {},
+  closeModal: () => {},
+});
+
+const MODALS: Record<string, ComponentType<ModalProps>> = {
```

Comment:
I don't think we should need to have this sort of type. FYI. Both EOLModal and ReleaseModal components shouldn't be imported here. The provider should receive the modal component as a children. This feels a bit hacky at best. This Provider should be generic IMO, and allow you to pass the Modal itself as children and then hook into it as a HOC.

Reply:
That exists because, if the server passed the client component to the provider, it caused Next.js server/client errors.

Although, we could take another stab at it

Reply:
i moved around some of this via https://github.com/nodejs/nodejs.org/pull/7990/commits/40df1abfcf62c060b0663f95ac24041e15e3b84b but i welcome feedback